sklearn/metrics/_plot/precision_recall_curve.py

```diff
@@ -325,6 +335,12 @@ def from_predictions(
         """
         check_matplotlib_support(f"{cls.__name__}.from_predictions")
 
+        target_type = type_of_target(y_true)
+        if target_type != "binary":
+            raise ValueError(
+                f"The target y is not binary. Got {target_type} type of target."
+            )
```

Comment:
We do this check in `_binary_clf_curve`:

https://github.com/scikit-learn/scikit-learn/blob/845b1fac9e8bcfc146e7b38119ce6c500e00a10b/sklearn/metrics/_ranking.py#L729-L731